src/components/cards/collaborations/AddCollaboratorButton.jsx

```diff
@@ -0,0 +1,232 @@
+import React, { useState } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import AddIcon from '@material-ui/icons/Add';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import TextField from '@material-ui/core/TextField';
+import SearchIcon from '@material-ui/icons/Search';
+
+import Button from '../../Button';
+import CustomAlert from '../../Alert';
+import DataDisplay from '../../dataDisplays/DataDisplay';
+import StandardDialog from '../../StandardDialog';
+import Text from '../../Text';
+import useGetUsers from '../../../models/users/useGetUsers';
+import { cellRendererTypes } from '../../dataDisplays/cellRenderers';
+import { getHighestRoleLabelId } from '../../../utils/roleUtils';
+
+function filterCollaborator(
+  potentialCollaborator,
+  searchTerm,
+  userGuid,
+  currentCollaboratorGuids,
+) {
+  // If there is no search term, no users should be returned
+  if (!searchTerm) return false;
+
+  // The user should not be able to collaborate with themself
+  if (potentialCollaborator?.guid === userGuid) return false;
+
+  // No internal or deactivated users should be available for collaboration
+  const isInternal = potentialCollaborator?.is_internal || false;
+  const isActive = potentialCollaborator?.is_active || false;
+  if (isInternal || !isActive) return false;
+
+  // No users that the user is already collaborating with should be available
+  // for collaboration
+  if (currentCollaboratorGuids.includes(potentialCollaborator?.guid))
+    return false;
+
+  // User name and email are filterable fields
+  const lowerCaseSearchTerm = searchTerm.toLowerCase();
```

Comment:
We don't use `trim()` on the filter term in DataDisplay, so I did it for consistency. It can also be useful for the filtering though:
My user list includes two users, "user one" and "one user". If I search for "one", both users will be included in the result. However, if I search for " one" with a leading space, only "user one" will be included, which is hopefully what I wanted.